### PR TITLE
Modernize forms #2591829

### PR DESCRIPTION
**Pull request by czigor**
https://www.drupal.org/node/2591829

### CODE REVIEW

**Comment by czigor** [comment]
Other forms are coming soon.

**Comment by bojanz** [comment]
Payment info won't be translatable, so we shouldn't add language settings there.

For other forms, make sure $language_configuration is $languageConfiguration.

**Comment by czigor** [comment]
TaxTypeForm seems a bit weird, should I touch it? For id it has  a textfield instead of machine_name and I could not find a TaxType entity. Otherwise I'm ready.

**Comment by bojanz** [comment]
No, you don't need to touch that one.

File: modules/product/src/Form/ProductTypeForm.php [hunk 1]
```diff
@@ -83,7 +86,27 @@ public function form(array $form, FormStateInterface $form_state) {
       '#description' => t('Products of this type represent digital services.')
     ];
 
-    return $form;
+    if ($this->moduleHandler->moduleExists('language')) {
+      $form['language'] = array(
```
Comment (bojanz) [inline modules/product/src/Form/ProductTypeForm.php:90]: Can we use the short array syntax here? [ instead of array(

**Comment by bojanz** [comment]
Committed as 0e675b7e5e49ddeafeb7b67975129a1f402e2238, made the short array syntax change manually, as well as the #disabled fix (using $type->isLocked()) that was forgotten.

**Pull request closed without merging**